cmd/peggo/ethrpc_manager.go

```diff
@@ -0,0 +1,138 @@
+package peggo
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"os"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/knadh/koanf"
+	"github.com/pkg/errors"
+)
+
+type EthRPCManager struct {
+	currentEndpoint int // index (in the slice of configured RPC endpoints) of most recent endpoint used
+	client          *rpc.Client
+	konfig          *koanf.Koanf
+}
+
+// creates an instance of EthRPCManager with a given konfig.
+func NewEthRPCManager(konfig *koanf.Koanf) *EthRPCManager {
```

Comment:
Would it be possible for there to be concurrent use of an `EthRPCManager` instance?

Reply:
Use of the embedded clients themselves should be concurrency-safe, but concurrent redialing / closing would need to be worked on - for example if one goroutine closed the client (perhaps as a result of an error in a grpc call) while another was doing something, the latter could geta very unexpected "connection was closed" type of error.

Or if two concurrent redialing loops were active, a whole lot could happen. It should be possible to add a lock on DialNext() in particular before merging if we move forward with this.